Showcase-iOSTests/FirebaseDataBase/DatabaseReferenceTests.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  DatabaseReferenceTests.swift
+//  Showcase-iOSTests
+//
+//  Created by Kagiso Mohajane on 2018/04/19.
+//  Copyright © 2018 DVT. All rights reserved.
+//
+
+import XCTest
+import Cuckoo
+@testable import Showcase_iOS
+
+class DatabaseReferenceTests: XCTestCase {
+    
+    let mockDatabaseReference = MockDataReferenceable()
+    let mockPath = ""
+    var systemUnderTest : FirebaseRetrieverableImplementation!
+    
+    override func setUp() {
+        super.setUp()
+        systemUnderTest  = FirebaseRetrieverableImplementation(reference: mockDatabaseReference)
+    }
+    
```

Comment:
you need to write more tests here. What happens when it is nil ? what is your error case? you need to not just cover the positive cases, the negatives also.